src/pcp.rs

```diff
@@ -113,19 +113,22 @@ pub enum PcpError {
 
 /// A value of a certain type. Implementations of this trait specify an arithmetic circuit that
 /// determines whether a given value is valid.
-pub trait Value:
-    Sized
-    + PartialEq
-    + Eq
-    + Debug
-    + for<'a> TryFrom<(<Self as Value>::Param, &'a [Self::Field]), Error = PcpError>
-{
+pub trait Value: Sized + PartialEq + Eq + Debug {
     /// The finite field used for this type.
     type Field: FieldElement;
 
     /// Parameters used to construct a value of this type from a vector of field elements.
     type Param: Clone + Debug;
 
+    /// Constructs an instance of a value from an additive share of the input (`data`) and type
+    /// parameter (`param`). For some types, the validity circuit may depend on the total number of
+    /// shares (`num_shares`).
+    fn new_share(
+        data: Vec<Self::Field>,
```

Comment:
```suggestion
        data: &[Self::Field],
```
More flexible than `Vec` and I don't think we need to consume the data.

Reply:
Actually, the data is always consumed. E.g., see the implementation of `Sum`.

Reply:
Yeah, seems like you either have to force the caller to allocate a `Vec`, or you have to force a `to_vec` or a `clone` in the various constructors. Let's keep this as-is.